src/app/grid/grid-validator-service-extended/grid-validator-service-extended.component.ts

```diff
@@ -55,46 +53,37 @@ export class GridValidatorServiceExtendedComponent {
      * Bind this handler to `cellEdit` output of the grid
      * in order to cancel cell editing in case the submitted
      * value is invalid.
-     * 
-     * @param evt 
+     *
+     * @param evt
      */
+
     public cellEdit(evt) {
-        if (!evt.isValid) {
+        if (!evt.valid) {
             evt.cancel = true;
         }
     }
 
     public commit() {
-        const invalidTransactions = this.grid.transactions.getInvalidTransactionLog();
+        const invalidTransactions = this.grid.validation.getInvalid();
         if (invalidTransactions.length > 0) {
             if (confirm('You\'re commiting invalid transactions. Are you sure?')) {
-                this.grid.transactions.commit(this.transactionData);
+                this.grid.validation.clear();
             }
         } else {
-            this.grid.transactions.commit(this.transactionData);
+            this.grid.validation.clear();
         }
     }
 
-    private _testFax(): ValidatorFn {
-        return (control: AbstractControl): ValidationErrors | null => {
-            const match = this._regexValidationFunction(control.value, /\+\d{1}\-(?!0)(\d{3})\-(\d{3})\-(\d{4})\b/i);
-            return !match ? { forbiddenFax: { value: control.value } } : null;
-        }
-    }
-
-    private _regexValidationFunction(value: any, rgx: RegExp): boolean {
-        const match = rgx?.test(value);
-        return !!match;
-    }
-
-    private _testDateRecord(thresholdVal?: any): ValidatorFn {
+    public _testDateRecord(thresholdVal?: any): ValidatorFn {
         return (control: AbstractControl): ValidationErrors | null => {
             const date = control.value;
-            const exceedingThreshold = !!thresholdVal ? date < thresholdVal : false;
-            if (!exceedingThreshold) {
-                return date < new Date() ? null : { forbiddenDate: { value: control.value } };
+            if(date > new Date()){
```

Comment:
This could be split in 2 validators. One that check if the date is in the future(pastDateValidator) and one that does the multi-field validation between created_on and last_activity fields (datesThresholdValidator). No need to have 1 validator that does 2 unrelated checks.